src/fileops/davmeta.cpp

```diff
@@ -565,6 +566,83 @@ static bool is_swift_operation(IOChainContext & context){
     return context._reqparams->getProtocol() == RequestProtocol::Swift;
 }
 
+static void swiftStatMapper(Context& context, const RequestParams* params, const Uri & uri, struct StatInfo& st_info) {
+    const std::string scope = "Davix::swiftStatMapper";
+    DavixError * tmp_err=NULL;
+    HeadRequest req(context, uri, &tmp_err);
+
+    // we need to modify it, hence copy
+    RequestParams p(params);
+
+    if(tmp_err == NULL) {
+        req.setParameters(p);
+        req.executeRequest(&tmp_err);
+        const int code = req.getRequestCode();
+
+        if(code == 404) {
+            DavixError::clearError(&tmp_err);
+            // try to "list" target resource and see if there is anything inside it, if there is, then it's a directory
+            Uri new_url = Swift::swiftUriTransformer(uri, p, true);
+
+            const int operation_timeout = p.getOperationTimeout()->tv_sec;
```

Comment:
Is this really needed?

S3 uses a DirHandle to judge whether the path points to a directory or something else.
The timeout variable there is used to stop the dir parsing in a timely manner.

Here, you're using a GET request, which has an internal timeout.